Assets/Editor/Dialogue Editor/GraphSaveUtility.cs

```diff
@@ -102,7 +102,28 @@ private void LinkNodes(Port port1, Port port2) {
 
     private void CreateNodes() {
         foreach(var nodeData in _containerCache.DialogueNodeData) {
-            var tempNode = _targetGraphView.CreateDialogueNode(nodeData.DialogueText);
+            DialogueNode tempNode;
+            if (nodeData.DialogueText == DialogueConstants.CheckerNodeName)
+            {
+                tempNode = _targetGraphView.CreateCheckerDialogueNode();
+            }
+            else if (nodeData.DialogueText == DialogueConstants.EventNodeName)
+            {
+                tempNode = _targetGraphView.CreateEventTriggerNode();
+            }
+            else if (nodeData.DialogueText == DialogueConstants.CharacterNodeName)
+            {
+                tempNode = _targetGraphView.CreateNewCharacterSpeakingNode();
+            }
+            else if (nodeData.DialogueText == DialogueConstants.EndNodeName)
+            {
+                tempNode = _targetGraphView.CreateEndHereNode();
+            }
+            else
+            {
+                tempNode = _targetGraphView.CreateDialogueNode(nodeData.DialogueText);
+            }   
```

Comment:
refactoring this into a switch or dictionary would be a lot better, something like this (i havent checked that the syntax is fully correct, so be careful when pasting)
```cs
tempNode = nodeData.DialogueText switch
{
  DialogueConstants.CheckerNodeName => _targetGraphView.CreateCheckerDialogueNode(),
  DialogueConstants.EventNodeName => _targetGraphView.CreateEventTriggerNode(),
  DialogueConstants.CharacterNodeName => _targetGraphView.CreateNewCharacterSpeakingNode(),
  DialogueConstants.EndNodeName => _targetGraphView.CreateEndHereNode(),
  _ => _targetGraphView.CreateDialogueNode(nodeData.DialogueText)
}
```